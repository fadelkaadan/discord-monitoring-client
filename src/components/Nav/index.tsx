import { NavLink } from "react-router-dom";
import styled from "styled-components";

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
`;

const NavItemWrapper = styled.div`
  width: 70%;
  max-width: 1000px;
  padding: 25px 0;
`;

const Nav = () => {
  return (
    <Wrapper>
      <NavItemWrapper>
        <NavLink
          style={({ isActive }) => ({
            color: isActive ? "#7289da" : "#fff",
            fontWeight: 700,
            marginRight: "20px",
            fontSize: "30px",
            textDecoration: "none",
          })}
          to={"/"}
        >
          Flagged messages
        </NavLink>
        <NavLink
          style={({ isActive }) => ({
            color: isActive ? "#7289da" : "#fff",
            fontWeight: 700,
            marginRight: "20px",
            fontSize: "30px",
            textDecoration: "none",
          })}
          to={"/censored-words"}
        >
          Censored words
        </NavLink>
      </NavItemWrapper>
    </Wrapper>
  );
};

export default Nav;
