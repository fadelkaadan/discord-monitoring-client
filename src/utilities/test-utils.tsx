import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import "@testing-library/jest-dom";
import "@testing-library/jest-dom/extend-expect";
import "@testing-library/user-event";

interface AllTheProvidersProps {
  children?: any;
}

const AllTheProviders = ({ children }: AllTheProvidersProps) => {
  return <Router>{children}</Router>;
};

const customRender = (ui: any, options?: any) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
