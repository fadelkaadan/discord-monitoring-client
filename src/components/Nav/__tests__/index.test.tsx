import { render, screen, RenderResult } from "../../../utilities/test-utils";

import Nav from "../index";

describe("Nav", () => {
  it("Should match the snapshot", () => {
    const { container }: RenderResult = render(<Nav />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("Should render links in the nav", () => {
    render(<Nav />);
    const linkOne: HTMLElement = screen.getByRole("link", {
      name: /Flagged messages/i,
    });
    const linkTwo: HTMLElement = screen.getByRole("link", {
      name: /Censored words/i,
    });
    expect(linkOne).toBeInTheDocument();
    expect(linkTwo).toBeInTheDocument();
  });
});
