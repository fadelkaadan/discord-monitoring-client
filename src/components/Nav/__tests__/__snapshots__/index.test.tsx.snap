// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Should match the snapshot 1`] = `
<div
  class="sc-bdvvtL hYwHED"
>
  <div
    class="sc-gsDKAQ BA-dEd"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
      style="color: rgb(114, 137, 218); font-weight: 700; margin-right: 20px; font-size: 30px; text-decoration: none;"
    >
      Flagged messages
    </a>
    <a
      class=""
      href="/censored-words"
      style="color: rgb(255, 255, 255); font-weight: 700; margin-right: 20px; font-size: 30px; text-decoration: none;"
    >
      Censored words
    </a>
  </div>
</div>
`;
